applicationinsights/django/middleware.py

```diff
@@ -0,0 +1,266 @@
+
+    # Pre-1.10 handler
+    def process_request(self, request):
+        # Populate context object onto request
+        addon = RequestAddon(self._client)
+        data = addon.request
+        context = addon.context
+        request.appinsights = addon
+
+        # Basic request properties
+        data.start_time = datetime.datetime.utcnow().isoformat() + "Z"
+        data.http_method = request.method
+        data.url = request.build_absolute_uri()
+        data.name = "%s %s" % (request.method, data.url)
+        context.operation.name = data.name
+        context.operation.id = data.id
+        context.location.ip = request.META.get('REMOTE_ADDR', '')
+        context.user.user_agent = request.META.get('HTTP_USER_AGENT', '')
+
+        # User
+        if request.user is not None and not request.user.is_anonymous and request.user.is_authenticated:
+            context.user.account_id = request.user.get_short_name()
+
+        # Run and time the request
+        addon.start_stopwatch()
+        return None
+
+    # Pre-1.10 handler
+    def process_response(self, request, response):
+        addon = request.appinsights
+        duration = addon.measure_duration()
+
+        data = addon.request
+        context = addon.context
+
+        # Fill in data from the response
+        data.duration = addon.measure_duration()
+        data.response_code = response.status_code
+        data.success = response.status_code < 400
+
+        # Submit and return
+        self._client.channel.write(data, context)
+        return response
+
+    # 1.10 and up...
+    def __call__(self, request):
+        self.process_request(request)
+        response = self.get_response(request)
+        self.process_response(request, response)
+        return response
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        if not hasattr(request, "appinsights"):
+            return None
+
+        data = request.appinsights.request
+        context = request.appinsights.context
+
+        # Operation name is the method + url by default (set in __call__),
+        # so if that's not set, then we'll use the view name.
+        if not self._settings.use_operation_url:
+            mod = inspect.getmodule(view_func)
+            name = view_func.__name__
+            if mod:
+                opname = "%s %s.%s" % (data.http_method, mod.__name__, name)
+            else:
+                opname = "%s %s" % (data.http_method, name)
+            data.name = opname
+            context.operation.name = opname
+
+        # Populate the properties with view arguments
+        if self._settings.record_view_arguments:
+            for i, arg in enumerate(view_args):
+                data.properties['view_arg_' + str(i)] = arg_to_str(arg)
+
+            for k, v in view_kwargs.items():
+                data.properties['view_arg_' + k] = arg_to_str(v)
+
+        return None
+
+    def process_exception(self, request, exception):
```

Comment:
Node.js allows users to configure if they want automatic request/exceptions/dependencies as separate options. Maybe we should allow that here? (in case you want requests but no exceptions, etc)